docs/PLUGIN.md

```diff
@@ -0,0 +1,314 @@
+            exit 1
+          fi
+```
+
+### Multi-Component Plugin with Direct List
+```yaml
+plugin_name: "multi-disk-check"
+type: "component_list"
+component_list:
+  - "root/auto:/"
+  - "home/auto:/home"
+  - "var/auto:/var"
+health_state_plugin:
+  steps:
+    - name: "check-disk-space"
+      run_bash_script:
+        content_type: "plaintext"
+        script: |
+          #!/bin/bash
+          df -h ${PAR} | grep ${NAME}
+          if [ $? -eq 0 ]; then
+            echo "Disk space OK for ${NAME}"
+            exit 0
+          else
+            echo "Disk space check failed for ${NAME}"
+            exit 1
+          fi
+```
+
+### Multi-Component Plugin with File
+```yaml
+plugin_name: "multi-disk-check"
+type: "component_list"
+component_list_file: "/etc/gpud/disk_checks.txt"
+health_state_plugin:
+  steps:
+    - name: "check-disk-space"
+      run_bash_script:
+        content_type: "plaintext"
+        script: |
+          #!/bin/bash
+          df -h ${PAR} | grep ${NAME}
+          if [ $? -eq 0 ]; then
+            echo "Disk space OK for ${NAME}"
+            exit 0
+          else
+            echo "Disk space check failed for ${NAME}"
+            exit 1
+          fi
+```
+
+### Init Plugin
+```yaml
+plugin_name: "setup-monitoring"
+type: "init"
+timeout: 5m
+health_state_plugin:
+  steps:
+    - name: "install-tools"
+      run_bash_script:
+        content_type: "plaintext"
+        script: |
+          #!/bin/bash
+          apt-get update
+          apt-get install -y monitoring-tools
+          systemctl enable monitoring-tools
+          systemctl start monitoring-tools
+```
+
+## Best Practices
+
+1. **Script Content**:
+   - Use `content_type: "plaintext"` for readability
+   - Use `content_type: "base64"` for complex scripts with special characters
+   - Always include proper error handling
+   - Use exit codes appropriately (0 for success, non-zero for failure)
+
+2. **Timeouts and Intervals**:
+   - Set appropriate timeouts based on script complexity
+   - Use intervals >= 1 minute for component plugins
+   - Consider system load when setting intervals
+
+3. **Component Lists**:
+   - Use `component_list_file` for large numbers of components
+   - Keep component names descriptive and unique
+   - Use parameters to make scripts more flexible
+
+4. **Error Handling**:
+   - Always check for errors in your scripts
+   - Provide meaningful error messages
+   - Use appropriate exit codes
+
+## Integration with GPUd
+
+Plugins integrate with GPUd's component system:
+- They appear in the component list
+- Their health states are tracked
+- They can be queried through the API
+- They participate in the overall system health status
+
+## API Access
+
+Plugins can be accessed through GPUd's API:
+- List all plugins: `GET /v1/components/custom-plugin`
+- Get plugin status: `GET /v1/states?component=<plugin_name>`
+- Trigger manual check: `POST /v1/components/check?componentName=<plugin_name>` 
```

Comment:
```suggestion
- Trigger manual check: `GET /v1/components/trigger-check?componentName=<plugin_name>` 
```

For example,

> curl -s -kL https://localhost:15132/v1/components/trigger-check?componentName=automatic-exit-1 | jq 

https://github.com/leptonai/gpud/blob/4cbd7434e79d26aade05d127562c9d425cb49cce/pkg/server/handlers_components.go#L134-L151